### PR TITLE
doc: Changelog v0.5.0

### PR DESCRIPTION
**Pull request by heinezen**
Ready for release :rocket: 

resolves https://github.com/SFTtech/openage/issues/1518

### CODE REVIEW

**Review by TheJJ — DISMISSED** [review]

**Comment by zoli111** [comment]
I found two small problems in the changelog:
- Mako should be added as a new dependency
- A typo: `1 new prenter demo`

**Comment by heinezen** [comment]
@zoli111 Thank you, fixed!

**Review by TheJJ — APPROVED** [review]

File: doc/changelogs/engine/v0.5.0.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+- Mark old game simulation implementation to legacy
+- Update macOS build instructions
+- Update dependencies
+  - Python >=3.9
+  - Cython >=0.29.32
+  - gcc >= 10
+  - clang >=10
+  - qt6 >= 6.2
+- Use Github Actions v3
```
[the remark this answers is not in the document]
Comment (heinezen) [reply doc/changelogs/engine/v0.5.0.md:111]: I moved this to https://github.com/SFTtech/openage/pull/1550 :)

File: doc/changelogs/engine/v0.5.0.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+- macOS build on ARM architecture
+- Python version detection for Python>=3.10
+- Replace Python `distutils` with something that isn't deprecated
+- Compatibility with Cython 3.0
+- Fix a lot of horrible typos as usual
+
+## Full commit log
+
+https://github.com/SFTtech/openage/compare/v0.4.1...v0.5.0
```
[the remark this answers is not in the document]
Comment (heinezen) [reply doc/changelogs/engine/v0.5.0.md:148]: Next one won't be as huge :laughing:

File: doc/changelogs/engine/v0.5.0.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
+# [0.5.0] - 2023-01-01
```
Comment (TheJJ) [inline doc/changelogs/engine/v0.5.0.md:1]: date - we could just leave it out?

**Review by TheJJ — APPROVED** [review]

File: doc/changelogs/engine/v0.5.0.md (hunk 3 above)
Comment (heinezen) [reply doc/changelogs/engine/v0.5.0.md]: fixed

**Review by TheJJ — APPROVED** [review]

**Pull request merged**